$OpenBSD$

Add a compile-time option to make Tor browser look for its data
directory in the user's home directory and turn it on for OpenBSD.
Also fold in nearly identical logic for defined(ANDROID) in the hope
this makes the patch easier to upstream.
--- xpcom/io/TorFileUtils.cpp.orig	Tue Aug 23 14:52:47 2016
+++ xpcom/io/TorFileUtils.cpp	Tue Sep  6 13:36:05 2016
@@ -6,6 +6,10 @@
 
 #include "TorFileUtils.h"
 
+#if defined(ANDROID) || defined(__OpenBSD__)
+# define TOR_BROWSER_DATA_IN_HOME_DIR 1
+#endif /* __OpenBSD__ */
+
 static nsresult GetAppRootDir(nsIFile *aExeFile, nsIFile** aFile);
 
 //-----------------------------------------------------------------------------
@@ -14,9 +18,20 @@ TorBrowser_GetUserDataDir(nsIFile *aExeFile, nsIFile**
 {
   NS_ENSURE_ARG_POINTER(aFile);
   nsCOMPtr<nsIFile> tbDataDir;
-
-#ifdef TOR_BROWSER_DATA_OUTSIDE_APP_DIR
   nsAutoCString tbDataLeafName(NS_LITERAL_CSTRING("TorBrowser-Data"));
+
+#if defined(TOR_BROWSER_DATA_IN_HOME_DIR)
+  const char *homeDir = getenv("HOME");
+  if (!homeDir || !*homeDir)
+    return NS_ERROR_FAILURE;
+  nsresult rv = NS_NewNativeLocalFile(nsDependentCString(homeDir), true,
+                                      getter_AddRefs(tbDataDir));
+  NS_ENSURE_SUCCESS(rv, rv);
+#if defined(__OpenBSD__)
+  rv = tbDataDir->AppendNative(tbDataLeafName);
+  NS_ENSURE_SUCCESS(rv, rv);
+#endif /* __OpenBSD__ */
+#elif defined(TOR_BROWSER_DATA_OUTSIDE_APP_DIR)
   nsCOMPtr<nsIFile> appRootDir;
   nsresult rv = GetAppRootDir(aExeFile, getter_AddRefs(appRootDir));
   NS_ENSURE_SUCCESS(rv, rv);
@@ -80,14 +95,6 @@ TorBrowser_GetUserDataDir(nsIFile *aExeFile, nsIFile**
     NS_ENSURE_SUCCESS(rv, rv);
   }
 #endif
-
-#elif defined(ANDROID)
-  // Orfox stores data in the app home directory.
-  const char* homeDir = getenv("HOME");
-  if (!homeDir || !*homeDir)
-    return NS_ERROR_FAILURE;
-  nsresult rv = NS_NewNativeLocalFile(nsDependentCString(homeDir), true,
-                                      getter_AddRefs(tbDataDir));
 #else
   // User data is embedded within the application directory (i.e.,
   // TOR_BROWSER_DATA_OUTSIDE_APP_DIR is not defined).
